pkg/datarepair/repairer/repairer.go

```diff
@@ -98,6 +98,10 @@ func (service *Service) process(ctx context.Context) error {
 		if err != nil {
 			zap.L().Error("Repair failed", zap.Error(err))
 		}
+		err = service.queue.Delete(ctx, seg)
```

Comment:
Should we delete the segment from the queue if the repair process fail?

Reply:
probably not- good catch. instead I think we need to update the `attempted` timestamp.

Reply:
If we don't delete it, we probably need another field in the database to order the results of repairQueue.Select() by, to move it to the back of the queue.  What we don't want is the repair queue getting 'stuck' trying to repair something it can't.  I'm not quite sure where the 'irreparable' logic is, or why we don't see it here.

Reply:
I thought that field would be the `attempted` timestamp right? but I'm guessing the logic for moving it back in the queue hasn't been added yet (I haven't checked)